File: benchmark/interactive_function/Metrix_multiplication/Metrix.py
import time

def main():
    n = 1000
    matrix_a = [float(i % n) for i in range(n * n)]
    matrix_b = [float(i // n) for i in range(n * n)]
    result = [0.0] * (n * n)

    print(f"Starting Matrix Multiplication (Python - Flat): {n}x{n}")
    # start = time.perf_counter()

    for i in range(n):
        for j in range(n):
            sum_val = 0.0
            for k in range(n):
                sum_val += matrix_a[i * n + k] * matrix_b[k * n + j]
            result[i * n + j] = sum_val

    # end = time.perf_counter()
    print(f"Sample Result [0]: {result[0]:.6f}")
    # print(f"Time: {end - start:.4f} sec")

if __name__ == "__main__":
    main()

File: benchmark/interactive_function/Metrix_multiplication/Metrix.java
public class Metrix {
    public static void main(String[] args) {
        int n = 1000;
        double[] a = new double[n * n];
        double[] b = new double[n * n];
        double[] result = new double[n * n];

        // Initialize matrices
        for (int i = 0; i < n * n; i++) {
            a[i] = i % n;
            b[i] = i / n;
        }

        System.out.println("Starting Matrix Multiplication (Java - Flat): " + n + "x" + n);
        // long start = System.currentTimeMillis();

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                double sum = 0;
                for (int k = 0; k < n; k++) {
                    // สูตร: (row * N) + col
                    sum += a[i * n + k] * b[k * n + j];
                }
                result[i * n + j] = sum;
            }
        }

        // long end = System.currentTimeMillis();
        System.out.printf("Sample Result [0]: %.2f\n", result[0]);
        // System.out.println("Time: " + (end - start) / 1000.0 + " sec");
    }
}